search/collector.go

```diff
@@ -30,3 +30,24 @@ type Collector interface {
 	SetFacetsBuilder(facetsBuilder *FacetsBuilder)
 	FacetResults() FacetResults
 }
+
+// DocumentMatchHandler is the type of document match callback
+// bleve will invoke during the search.
+// Bleve will indicate the completion of ongoing search, by passing
+// nil value for the document match callback eventually.
+// It is recommended that the application takes a copy of the
+// hit/documentMatch if they wish to own it or need prolonged
+// access to it.
+type DocumentMatchHandler func(hit *DocumentMatch) error
```

Comment:
Should document what happens when the handler returns an error (looks like it will stop the search early, which would be useful in some cases)

Reply:
done!